scripts/bozzcoin.js

```diff
@@ -0,0 +1,203 @@
+      case 'cycled': return 'cycle';
+      default: return verb;
+    }
+  }
+
+  function bozzExercisesFrequentEnough() {
+    const thresholdInMilliSecs = bozziplierThreshold * 86400000;
+    const bozzLastExercised = robot.brain.get('bozzLastExercised');
+    return (new Date().getTime() - new Date(bozzLastExercised).getTime()) > thresholdInMilliSecs;
+  }
+
+  // returns true if bozziplier is reset
+  function updateBozziplier(username) {
+    if (username === 'rurouni') {
+      robot.brain.set('bozzLastExercised', new Date());
+      robot.brain.set('bozziplier', 1);
+      return true;
+    }
+    const bozzLastExercised = robot.brain.get('bozzLastExercised');
+    if (!bozzLastExercised || !bozzExercisesFrequentEnough()) {
+      let bozziplier = robot.brain.get('bozziplier');
+      if (!bozziplier) {
+        bozziplier = 1;
+        robot.brain.set('bozzLastExercised', new Date());
+      }
+      const newBozziplier = bozziplier * bozziplierDecay;
+      robot.brain.set('bozziplier', newBozziplier);
+    }
+    return false;
+  }
+
+  function convertToBozzcoin(reps, exerciseType) {
+    return Math.round(reps * earnRate(exerciseType) * robot.brain.get('bozziplier'));
+  }
+
+  function convertToReps(bozzcoin, exerciseType) {
+    return bozzcoin / earnRate(exerciseType);
+  }
+
+  function cheatDayPrices(cheatType) {
+    switch (cheatType) {
+      case 'prata': return 12000;
+      case 'starbucks': return 5000;
+      case 'macs': return 7000;
+      default: return 0;
+    }
+  }
+
+  robot.respond(new RegExp('set bozzcoins to (\\d+)', 'i'), (res) => {
+    if (res.message.user.name === 'lockheed') {
+      const bozzcoinBalance = Number.parseInt(res.match[1], 10);
+      robot.brain.set('bozzcoinBalance', bozzcoinBalance);
+      robot.brain.set('bozzcoinTracker', {});
+      res.send(`Bozzcoin balance reset to ${bozzcoinBalance} :bozzcoin:. Tracker reset.`);
+    }
+  });
+
+  robot.respond(/bozzcoins\?/i, (res) => {
+    const bozzcoinBalance = robot.brain.get('bozzcoinBalance');
+    res.send(`*${bozzcoinBalance}* :bozzcoin:`);
+    res.send(bozzcoinSummaries());
+  });
+
+  robot.respond(new RegExp('did (-?\\d+) (pullups|pushups|situps|squats|lunges|racket steps)', 'i'), (res) => {
+    let repsByUser;
+    const username = res.message.user.name;
+    const repsDone = Number.parseInt(res.match[1], 10);
+    const exerciseType = res.match[2];
+    const bozzcoinTracker = robot.brain.get('bozzcoinTracker');
+    if (username in bozzcoinTracker) {
+      repsByUser = convertToReps(bozzcoinTracker[username], exerciseType);
+    } else {
+      repsByUser = 0;
+    }
+    if (repsDone < (-1 * repsByUser)) {
+      res.send('You can\'t undo more than you have done');
+      return;
+    }
+    switch (exerciseType) {
+      case 'pushups': case 'situps': case 'squats': case 'lunges':
+        if (repsDone > 60) {
+          res.send(`Show the team you can do more ${exerciseType} than :commando:`);
+          return;
```

Comment:
Is it intended to just show the message without updating the bozz coins?

Reply:
This code already exists. It's to set upper bound for exercise types. If you're clocking in more than the upper bound, it sends that 'error' message.

Reply:
Maybe we should raise the upper bound. 60 situps seems do-able in one go. Also, the message gives no indication that the bozz coins are not increased.

Reply:
If bozz coins are increased, there will be a message that says so

Reply:
The bound can be discussed in the slack channel and modified in another PR.

Given the message patterns, imo it's quite clear that nothing happened since every successful clocking tells you info like bozzcoins earned, bozziplier, and current total bozzcoins. If this is felt strongly we can discuss in the slack channel for another PR.

Reply:
Ok, fair enough.